subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -0,0 +1,1348 @@
+/*
+ * Copyright (client) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <net/mqtt.h>
+#include <net/socket.h>
+#include <stdio.h>
+#include <net/azure_iot_hub.h>
+#include <settings/settings.h>
+
+#include "azure_iot_hub_dps.h"
+
+#if defined(CONFIG_AZURE_FOTA)
+#include <net/azure_fota.h>
+#endif
+
+#if defined(CONFIG_BOARD_QEMU_X86)
+#include "certificates.h"
+#endif
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(azure_iot_hub, CONFIG_AZURE_IOT_HUB_LOG_LEVEL);
+#define AZURE_IOT_HUB_AF_FAMILY AF_INET
+
+#define USER_NAME_STATIC	CONFIG_AZURE_IOT_HUB_HOSTNAME "/"	\
+				CONFIG_AZURE_IOT_HUB_DEVICE_ID		\
+				"/?api-version=2018-06-30"
+/* User name when connecting to Azure IoT hub is on the form
+ *	<IoT Hub hostname>/<device ID>/?api-version=2018-06-30
+ */
+#define USER_NAME_TEMPLATE	"%s/%s/?api-version=2018-06-30"
+
+#define TOPIC_DEVICEBOUND	"devices/%s/messages/devicebound/#"
+#define TOPIC_TWIN_DESIRED	"$iothub/twin/PATCH/properties/desired/#"
+#define TOPIC_TWIN_RES		"$iothub/twin/res/#"
+#define TOPIC_TWIN_REPORT	"$iothub/twin/PATCH/properties/reported/?$rid=%d"
+#define TOPIC_EVENTS		"devices/%s/messages/events/"
+#define TOPIC_TWIN_REQUEST	"$iothub/twin/GET/?$rid=%d"
+#define TOPIC_DIRECT_METHODS	"$iothub/methods/POST/#"
+#define TOPIC_DIRECT_METHOD_RES	"$iothub/methods/res/%d/?$rid=%d"
+
+#define DPS_USER_NAME		CONFIG_AZURE_IOT_HUB_DPS_ID_SCOPE \
+				"/registrations/%s/api-version=2019-03-31"
+
+#define FOTA_PROGRESS_REPORT	"{\"firmware\":{\"status\":\"%s\",\"dl\":%d}}"
```

Comment:
`dl` propery should only be included if download progress reporting is enabled.

Reply:
Yep. I thought I had added a Kconfig option to leave this out altogether, but apparently I didn't.

Reply:
Maybe I have not read the source correctly, when this is used.
I assumed it was used also to report start and finish of a FOTA.

Reply:
It's used for that, yes. I meant that I thought I had added the option to leave out the `dl` field and only report states like `downloading` and `applying`, but seems `dl` is always added, and the hardcoded value on line 49 makes the download process being sent on every 20% of progress, which doesn't add a whole lot of value. 

I'll make if configurable whether progress reports shall be sent at all, and if so, at which progress increments [%] it should be sent.